### PR TITLE
Bug 1628759 - Hide bugs in See Also fields from users without access

### PR DESCRIPTION
**Pull request by dklawren**

### CODE REVIEW

**Comment by dklawren** [comment]
Not yet ready for review. Want to do some more testing.

**Comment by dklawren** [comment]
Ok ready for review. Verified that someone with out permission to see the bug referred to cannot remove the reference.

File: qa/t/webservice_bug_get_bugs.t [hunk 1]
```diff
@@ -121,6 +120,14 @@ sub post_success {
...
   }
 
+  # See also to a private bug should not display for the public bug
+  if (!$is_private_bug && !$is_private_user) {
+    ok(
+      !exists $bug->{see_also} || !@{$bug->{see_also}},
+      'See also to a private bug should not display for the public bug and normal user'
+    );
+  }
+
```
Comment (globau) [inline qa/t/webservice_bug_get_bugs.t:130]: I'd like to see a few more tests here to cover the whole matrix: public/public, public/private, private/public, and private/private

File: qa/t/webservice_bug_get.t [hunk 2]
```diff
@@ -124,6 +123,14 @@ sub post_success {
...
   }
 
+  # See also to a private bug should not display for the public bug
+  if (!$is_private_bug && !$is_private_user) {
+    ok(
+      !exists $bug->{see_also} || !@{$bug->{see_also}},
+      'See also to a private bug should not display for the public bug and normal user'
+    );
+  }
+
```
Comment (globau) [inline qa/t/webservice_bug_get.t:133]: I'd like to see a few more tests here to cover the whole matrix: public/public, public/private, private/public, and private/private

**Review by globau — CHANGES_REQUESTED** [review]
Users should be prevented from adding or removing bugs they cannot see to/from see-also, both via the API and at bug creation time.

**Review by globau — APPROVED** [review]

**Pull request merged**